### ISSUE TITLE
Consider switching RMSD sqrt guard from max(mse, eps) to mse + eps for smoother gradients

### ISSUE THREAD

**Issue by hunter-heidenreich** [hunter-heidenreich/Kabsch-Cookbook#137]
## Summary

The RMSD computation in all autodiff frameworks uses `sqrt(max(mse, eps))` to prevent `d/dx sqrt(0) = inf` in the backward pass. This works, but creates a **gradient discontinuity** at `mse = eps`:

- `mse < eps`: gradient = 0 (clamp active, mse ignored)
- `mse >= eps`: gradient = `1/(2*sqrt(mse))`

At the boundary, the gradient jumps from 0 to `1/(2*sqrt(eps))` (~1,581 for float32). This can cause oscillation during training when the loss is near convergence.

## Proposal

Switch to `sqrt(mse + eps)`, the standard deep learning pattern (used in LayerNorm, GroupNorm, L2 normalization, etc.):

```python
# Current
rmsd = sqrt(max(mse, eps))

# Proposed
rmsd = sqrt(mse + eps)
```

### Benefits
- **Smooth gradient everywhere**: `1/(2*sqrt(mse + eps))` -- no discontinuity
- **Same floor**: minimum RMSD = `sqrt(eps)` (identical to current)
- **Same gradient ceiling**: max gradient = `1/(2*sqrt(eps))` (identical to current)
- **Always nonzero gradient**: even below the floor, gradient responds to mse changes (proportionally tiny, but continuous)

### Tradeoffs
- Slight upward bias on forward RMSD for all values (adds eps inside sqrt). For float32, `sqrt(1.0 + 1.19e-7) - sqrt(1.0) ≈ 6e-8` -- negligible in practice.
- Behavioral change to every RMSD output, so needs test validation

## Affected files (16 sites)

All `kabsch`, `kabsch_umeyama`, `horn`, and `horn_with_scale` functions in:
- `pytorch/kabsch_svd_nd.py`, `pytorch/horn_quat_3d.py`
- `jax/kabsch_svd_nd.py`, `jax/horn_quat_3d.py`
- `tensorflow/kabsch_svd_nd.py`, `tensorflow/horn_quat_3d.py`
- `mlx/kabsch_svd_nd.py`, `mlx/horn_quat_3d.py`

NumPy is unaffected (forward-only, uses `a_min=0.0`).

## Context

Identified during PR #136 review. The `max` clamp was inherited from the original `1e-12` hardcoded value and carried forward into the dtype-aware `finfo(dtype).eps` refactor.
